ArrayList/ArrayList.cs

```diff
@@ -1,8 +1,580 @@
+                    }
+                }
+
+                return indexMinValue;
+            }
+
+            return -1;
+        }
+
+        /// <summary>
+        /// Return index max value of list.
+        /// </summary>
+        /// <returns>Index max value.</returns>
+        public int GetIndexMaxValue()
+        {
+            if (_array.Length > 0)
+            {
+                T maxValue = _array[0];
+                int indexMaxValue = 0;
+                for (int i = 0; i < Length; i++)
+                {
+                    if (_array[i].CompareTo(maxValue) == 1)
+                    {
+                        maxValue = _array[i];
+                        indexMaxValue = i;
+                    }
+                }
+
+                return indexMaxValue;
+            }
+
+            return -1;
+        }
+
+        /// <summary>
+        /// Adds an ArrayList to the end of the list.
+        /// </summary>
+        /// <param list="list">ArrayList.</param>
+        public void AddArrayList(ArrayList<T> list)
+        {
+            int lastIndex = Length;
+            AddArrayListByIndex(list, lastIndex);
+        }
+
+        /// <summary>
+        /// Adds an ArrayList to the beginning of the list.
+        /// </summary>
+        /// <param list="list">ArrayList.</param>
+        public void AddArrayListToStart(ArrayList<T> list)
+        {
+            int firstIndex = 0;
+            AddArrayListByIndex(list, firstIndex);
+        }
+
+        /// <summary>
+        /// Adds an ArrayList to the list by index.
+        /// </summary>
+        /// <param list="list">ArrayList.</param>
+        /// <param index="index">Index by which the value will be inserted.</param>
+        public void AddArrayListByIndex(ArrayList<T> list, int index)
+        {
+            if (index >= 0 && index <= Length)
+            {
+                Length += list.Length;
+                if (Length >= _array.Length)
+                {
+                    UpSize();
+                }
+
+                int n = list.Length;
+                for (int i = Length - 1; i >= index; i--)
+                {
+                    if (i + n < _array.Length)
+                    {
+                        _array[i + n] = _array[i];
+                    }
+                }
+
+                int count = 0;
+                for (int i = index; i < Length; i++)
+                {
+                    if (count < list.Length)
+                    {
+                        _array[i] = list[count++];
+                    }
+                }
+            }
+            else
+            {
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Sorts the list with inserts.
+        /// </summary>
+        /// <param isDescending="isDescending">If true, it sorts in descending order. If false, it sorts in ascending order.</param>
+        public void SortInsertion(bool isDescending)
```

Comment:
I think, you can remove `Insertion` from name

Reply:
ok